packages/material-ui/src/Stepper/Stepper.js

```diff
@@ -63,19 +63,27 @@ function Stepper(props) {
       connector: connectorProp,
     };
 
+    function getStepState(active, idx) {
+      if (active === idx) return { active: true };
+      if (!nonLinear && active > idx) return { completed: true };
+      if (!nonLinear && active < idx) return { disabled: true };
+      return {};
+    }
+
     const state = {
       active: false,
       completed: false,
       disabled: false,
+      ...getStepState(activeStep, index),
+    };
+    
+    /* State forwarded to alternativeLabel connector if applicable */
+    const prevStepState = {
+      active: false,
+      completed: false,
+      disabled: false,
+      ...((!controlProps.last && getStepState(activeStep, index + 1)) || {}),
```

Comment:
So let me just spell this out in case I misunderstood it. 
`If this is not the last step get the state of the next step and call this the step of the previous step`

Shouldn't this be `index - 1` then? 

Could you also move the boundary guard to `getStepState`? Inlining this looks complicated.

Reply:
definitely right. The logic was sound but the naming was off. I have fixed this.